### ISSUE TITLE
Protobuf flow changes and updates

### ISSUE THREAD

**Issue by SpicyLemon** [provenance-io/provenance#1339]
## Summary

We need to push the third_party protobuf stuff up more often so that we can properly update the `buf.lock` file allowing us to push our own protos up when we need to.

We also need to bring our protobuf related scripts and stuff up-to-date with more recent commands.

## Problem Definition

When we created `v1.14.0-rc1`, the protobuf push job failed. It failed because the `proto/buf.lock` file points to an old commit that doesn't have a file recently added. In order to update that file, though, we need to first push up our third_party stuff.

This highlights a major flow problem. In order to keep both the third_party and provenance stuff updated in `buf.build`, the third party stuff needs to be pushed up first, then we need to update the lock file before we can push up our stuff. Workflows can't be allowed to update files though, so it's something we need to do.

Additionally, `protoc` isn't a thing anymore and we need to not stop using it.

## Proposal

Update the github actions workflows related to the protobuf pushing. Currently, this is done in `releases.yaml` in its own job that's triggered when we tag a release.
1. Create a new job that pushes the third_party stuff any time a PR is merged into main that has changes in the third_party directory. This might be better placed in a new workflow just for that. 
2. Update the releases.yaml to only push the provenance protos. They should still only be pushed upon release.
3. See if there's a way to have a github action create an issue. If so, whenever this third_party stuff is pushed, have it create an issue to update the lock file. If not, update the release process documentation (in `CONTRIBUTING.md`) to include a step about update that lock file.

Also, take a look at what the SDK does for proto generation stuff and update our own scripts to be similar. They don't do the same thing with docs that we do though, so some research will be needed on how we can continue to have our documentation auto-generated from the protos.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned
